library/src/test/java/org/apache/apex/malhar/lib/fs/FSRecordReaderTest.java

```diff
@@ -89,15 +89,18 @@ public void testDelimitedRecords() throws Exception
     lma.prepareDAG(app, conf);
     LocalMode.Controller lc = lma.getController();
     lc.setHeartbeatMonitoringEnabled(true);
-    lc.runAsync();
+    lc.run(10 * 1000);
```

Comment:
Please implement proper test termination. This topic was discussed elsewhere before.